crates/autopilot/src/run.rs

```diff
@@ -448,17 +458,45 @@ pub async fn run(args: Arguments) {
 
     let persistence =
         infra::persistence::Persistence::new(args.s3.into().unwrap(), Arc::new(db.clone())).await;
-    let on_settlement_event_updater =
-        crate::on_settlement_event_updater::OnSettlementEventUpdater::new(
-            eth.clone(),
-            db.clone(),
+    let updater =
```

Comment:
not sure how to call this struct. slightly better is db_updater but still bad. The main functionality of this updater is to populate the auction-settlement relation in the database, or other way said, attach the newly observed settlement to one of the auctions that ran recently.